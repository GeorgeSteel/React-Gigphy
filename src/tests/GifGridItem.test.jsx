import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import GifGridItem from '../components/GifGridItem'

describe('GifGridItem Component Tests', () => {
  const gif = {
    id: '123',
    url: 'https://',
    title: 'something',
  }
  const wrapper = shallow(<GifGridItem {...gif} />)

  test('should show the component without issues', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should have a paragraph with the title', () => {
    const p = wrapper.find('p')
    expect(p.text().trim()).toBe(gif.title)
  })

  test('the image should have an url & alt from the props', () => {
    const img = wrapper.find('img')

    expect(img.prop('src')).toBe(gif.url)
    expect(img.prop('alt')).toBe(gif.title)
  })

  test('should have an animate class', () => {
    const div = wrapper.find('div')

    expect(div.prop('className').includes('animate__fadeInDown')).toBe(true)
  })
})
