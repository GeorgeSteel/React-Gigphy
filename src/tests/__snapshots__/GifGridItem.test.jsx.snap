// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Component Tests should show the component without issues 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="something"
    src="https://"
  />
  <p>
    something
  </p>
</div>
`;
